modules/shared/src/main/scala/org/tessellation/schema/transaction.scala

```diff
@@ -27,73 +25,65 @@ import monocle.macros.GenLens
 
 object transaction {
 
-  @derive(decoder, encoder, eqv, show)
+  @derive(decoder, encoder, order, show)
   @newtype
   case class TransactionAmount(value: PosLong)
 
   object TransactionAmount {
     implicit def toAmount(amount: TransactionAmount): Amount = Amount(amount.value)
   }
 
-  @derive(decoder, encoder, eqv, show)
+  @derive(decoder, encoder, order, show)
   @newtype
   case class TransactionFee(value: NonNegLong)
 
   object TransactionFee {
     implicit def toAmount(fee: TransactionFee): Amount = Amount(fee.value)
   }
 
-  @derive(decoder, encoder, show)
+  @derive(decoder, encoder, order, show)
   @newtype
   case class TransactionOrdinal(value: NonNegLong) {
     def next: TransactionOrdinal = TransactionOrdinal(value |+| 1L)
   }
 
   object TransactionOrdinal {
     val first: TransactionOrdinal = TransactionOrdinal(1L)
-
-    implicit val order: Order[TransactionOrdinal] = (x: TransactionOrdinal, y: TransactionOrdinal) =>
-      implicitly[Order[Long]].compare(x.coerce, y.coerce)
-
-    implicit val ordering: Ordering[TransactionOrdinal] = order.toOrdering
-
   }
 
-  @derive(decoder, encoder, eqv, show)
-  case class TransactionReference(hash: Hash, ordinal: TransactionOrdinal)
+  @derive(decoder, encoder, order, show)
+  case class TransactionReference(ordinal: TransactionOrdinal, hash: Hash)
 
   object TransactionReference {
-    val empty: TransactionReference = TransactionReference(Hash(""), TransactionOrdinal(0L))
+    val empty: TransactionReference = TransactionReference(TransactionOrdinal(0L), Hash("".padTo(64, '0')))
 
     val _Hash: Lens[TransactionReference, Hash] = GenLens[TransactionReference](_.hash)
     val _Ordinal: Lens[TransactionReference, TransactionOrdinal] = GenLens[TransactionReference](_.ordinal)
 
-    def of[F[_]: Async: SecurityProvider: KryoSerializer](transaction: Signed[Transaction]): F[TransactionReference] =
-      transaction.hashWithSignatureCheck.flatMap(_.liftTo[F]).map { hashed =>
-        TransactionReference(hashed.hash, hashed.ordinal)
-      }
+    def of[F[_]: Async: KryoSerializer](signedTransaction: Signed[Transaction]): F[TransactionReference] =
+      signedTransaction.value.hashF.map(TransactionReference(signedTransaction.ordinal, _))
 
   }
 
-  @derive(decoder, encoder, eqv, show)
+  @derive(decoder, encoder, order, show)
   @newtype
   case class TransactionSalt(value: Long)
 
-  @derive(decoder, encoder, eqv, show)
+  @derive(decoder, encoder, order, show)
   case class TransactionData(
     source: Address,
     destination: Address,
     amount: TransactionAmount,
     fee: TransactionFee
   )
 
-  @derive(decoder, encoder, eqv, show)
+  @derive(decoder, encoder, order, show)
```

Comment:
I think it can be misleading that it depends on the order of properties and it can be really easy to miss :|

Reply:
True. My intention was to have a sensible default order. However, I agree that it may lead to errors in the future if someone starts depending on the default order, while someone else changes the default order breaking the other places.